openedx/core/lib/xblock_utils/__init__.py

```diff
@@ -412,7 +412,12 @@ def add_staff_markup(user, disable_staff_debug_info, block, view, frag, context)
         'disable_staff_debug_info': disable_staff_debug_info,
     }
     if isinstance(block, ScorableXBlockMixin):
-        staff_context['max_problem_score'] = block.max_score()
+        try:
+            max_score = block.max_score()
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Avoid catching broad exception here, only catch the intended `LoncapaProblemError` here so that other exceptions don't catch up here and set the max_score to 0.

Reply:
It can be done, but the code where we are setting this is a generic xblock util code and is irrelevant to any specific block. LoncapaProblem being a specific type can introduce a bit of coupling in this piece of code. This can be debated a bit more.